lib/EasyPost/Service/CarrierMetadataService.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+namespace EasyPost\Service;
+
+use EasyPost\Http\Requestor;
+use EasyPost\Util\InternalUtil;
+
+/**
+ * CarrierMetadata service containing all the logic to make API calls.
+ */
+class CarrierMetadataService extends BaseService
+{
+    /**
+     * Get carrier metadata for all carriers on the EasyPost platform.
+     *
+     * @param array|null $carriers
+     * @param array|null $types
+     * @return mixed
+     */
+    public function retrieveCarrierMetadata($carriers = null, $types = null)
+    {
+        $url = '/metadata';
+        if (isset($carriers) || isset($types)) {
+            $url = "{$url}?";
+        }
+        if (isset($carriers)) {
+            $url = "{$url}carriers=" . join(',', $carriers);
+        }
+        if (isset($carriers) || isset($types)) {
+            $url = "{$url}&";
+        }
```

Comment:
```suggestion
        if (isset($carriers) && isset($types)) {
            $url = "{$url}&";
        }
```

Issue: We need to make a small tweak here to make it `&&` to handle all the edge cases. 

1) If the users only give `carriers` in the parameter but not `types`, it will append `&` at the end of query URL.

2) If the users only give `types` in the parameter but not `carrier`, the query URL will look something like this `/metadata?&types=xyz,xyz`